brave/brave-core
# Add Privacy Pass extension

This PR addresses https://github.com/brave/brave-browser/issues/1677 by adding Privacy Pass extension. Extension is added as a regular extension installed from the [Chrome store](https://chrome.google.com/webstore/detail/privacy-pass/ajhmfdgkijocedmfjonnpjfojldioehi), in a fashion identical to the IPFS Companion extension.

After https://github.com/brave/brave-browser/issues/2307 has been implemented this will allow users to have a hassle free experience in all contexts. For now it works in normal and incognito modes, but not in Tor mode.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:
1. Visit brave://settings/extensions and toggle Privacy Pass switch, a prompt as below will be shown if Privacy Pass extension is not currently installed.
<img width="1203" alt="Screenshot 2019-06-03 at 14 50 20" src="https://user-images.githubusercontent.com/7261674/58803093-f59ecd00-860e-11e9-83d3-5d2bdca0c18b.png">
2. Click "Cancel" in the prompt, the Privacy Pass switch should be changed back to off.
3. Toggle it on again, and click "Add extension" in the prompt to install it, Privacy Pass extension should now be installed and enabled.
4. Left click Privacy Pass icon opens current status report, as shown below
<img width="1201" alt="Screenshot 2019-06-03 at 14 53 39" src="https://user-images.githubusercontent.com/7261674/58803330-7cec4080-860f-11e9-9ae4-c3f83b77b800.png">
5. In brave://settings/extensions, toggle Privacy Pass switch off again and visit brave://extensions, Privacy Pass extension should be shown disabled.
6. Enable Privacy Pass switch toggle in brave://settings/extensions again,Privacy Pass extension should be shown enabled in brave://extensions.
7. Remove Privacy Pass extension in brave://extensions, Privacy Pass extension switch toggle in brave://settings/extensions should be shown off.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
